pkg/apis/core/types.go

```diff
@@ -635,6 +664,29 @@ type PersistentVolumeClaimStatus struct {
 	// +mapType=granular
 	// +optional
 	AllocatedResourceStatuses map[ResourceName]ClaimResourceStatus
+	// volumeAttributesClassName is the name of VolumeAttributesClass which is latest applied by
+	// the external-provisioner or external-resizer. This field is not set by default but when
+	// modification is complete, it's set same as VolumeAttributesClassName in spec.
+	// This is an alpha field and requires enabling VolumeAttributesClass feature.
+	// +featureGate=VolumeAttributesClassName
+	// +optional
+	VolumeAttributesClassName *string
+	// VolumeAttributesModifyStatus stores status of modification operation. It can be in any of following states:
```

Comment:
nit: instead of state, it would be better to call it "conditions". We don't want to imply that these are state machines: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

It would also be good to state that more conditions could be added in the future and how controllers should handle conditions they don't understand.

Reply:
The status struct of the pvc has already this field called `conditions`. Do you mean that we should add a new or more conditionTypes like `ControllerMidifyVolumeAttributes` instead of the VolumeAttributesModifyStatus field?

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L573

cc @sunnylovestiramisu

Reply:
Should it be ControllerModifyVolumeConditions?

Reply:
Discussed offline, we will not have the Pending state. Only in progress, failed and succeed.

Reply:
And I just see that Conditions is a list, which makes it harder to loop up a specific condition for only VAC. What is the general guideline on using multiple conditions? Nothing specific from the guide: [github.com/kubernetes-csi*](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)  @msau42

Reply:
Resources can have multiple conditions. The key thing to think about, is that conditions are not state machines. Each condition should be able to independently reason about, and you should be able to add new conditions and not break older clients that don't know about those conditions.

Reply:
I don't know what new conditionTypes we want to define. Can you give me some examples? In my mind, perhaps we just need one new conditionType, which is "VolumeAttributesClassMismatched". 

`InProgress`, `Failed`, and more are the corresponding reason for the conditionType `VolumeAttributesClassMismatched`.

What do you think? @msau42 @sunnylovestiramisu 

An another question is which component is responsible for setting the conditionType? pv controller, external CSI driver or both?

Reply:
Yes, the API conventions doc gives an example of Resizing as the conditon type, and inprogress/failed/succeeded could be reasons.

However, after discussing with @gnufied it looks like we have some inconsistency issues with volume resizing feature as well as Pod resizing feature, which don't use conditions. I started a discussion in slack so we can align with the other features.

Meanwhile, @gnufied and I brainstormed possible status structures assuming we go with a status approach instead of conditions. However, I do like the some of the metadata that's in conditions so this is a bit of a hybrid approach. Please take a look and let me know what you think:

https://gist.github.com/msau42/66cbdd43b9c2b5a5fb245b8de0f8338d

In this proposal, note that there is also an additional "targetClass" field. We added this to handle the case where someone may try to change the vac while we're still in the middle of changing it. We need to store what our current target is so that we can correctly account for quota and also know the correct class to set when the operation is complete.